server/test/unit/server/db/test_model.py

```diff
@@ -336,6 +330,23 @@ def test_pre_save_signal_already_has_storage_path(self, get_path):
         self.assertFalse(get_path.called)
         self.assertEqual(document._storage_path, '123')
 
+    @patch('pulp.server.db.model.FileStorage.get_path')
+    def test_set_file(self, get_path):
+        path = '/tmp/9876'
+        get_path.return_value = path
+        name = '1234'
+        unit = TestFileContentUnit.TestUnit()
+        unit.set_file(name)
+        get_path.assert_called_once_with(unit)
+        self.assertEqual(unit._storage_path, os.path.join(path, name))
+
+    def test_set_file_ignored(self):
+        path = '/tmp/9876'
+        unit = TestFileContentUnit.TestUnit()
+        unit._storage_path = path
+        unit.set_file('1234')
+        self.assertEqual(unit._storage_path, path)
```

Comment:
IMO, it would be good to assert that this situation raises an Exception, perhaps a RuntimeError?